### PR TITLE
roachpb: Remove the txn field from TransactionPushError

### PR DESCRIPTION
**Pull request by kkaneda**
Use the txn field in Error instead.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4366)

### CODE REVIEW

File: roachpb/errors.go [hunk 1]
```diff
@@ -363,10 +354,10 @@ func (e *TransactionPushError) Error() string {
 
 // message returns an error message.
 func (e *TransactionPushError) message(pErr *Error) string {
-	if e.Txn == nil {
+	if pErr.UnexposedTxn == nil {
```
Comment (bdarnell) [inline roachpb/errors.go]: All reads of `UnexposedTxn` should go through `GetTxn()`.

**Comment by bdarnell** [comment]
LGTM

File: roachpb/errors.go [hunk 2]
```diff
@@ -342,18 +342,9 @@ func NewTransactionAbortedError() *TransactionAbortedError {
 }
 
 // NewTransactionPushError initializes a new TransactionPushError.
-// Txn is the transaction which will be retried. Both arguments are copied.
-// Transactions.
-func NewTransactionPushError(txn, pusheeTxn Transaction) *TransactionPushError {
-	err := &TransactionPushError{PusheeTxn: pusheeTxn.Clone()}
-	if txn.ID != nil {
```
Comment (tschottdorf) [inline roachpb/errors.go]: This behaviour has changed, and that's bad: Now a non-transactional pusher will get a half-baked transaction error back.

Comment (tschottdorf) [inline roachpb/errors.go]: Oh, sorry. Nevermind, completely missed the point of this.

**Comment by tschottdorf** [comment]
LGTM (mod Ben's comment)

File: roachpb/errors.go (hunk 1 above)
Comment (kkaneda) [inline roachpb/errors.go]: Fixed.

**Comment by kkaneda** [comment]
Thanks for the reviews!

File: roachpb/errors.go (hunk 1 above)
Comment (tamird) [inline roachpb/errors.go]: looks like this slipped through unchanged =/

Comment (kkaneda) [inline roachpb/errors.go]: Hmm :( Sending a fix..

**Pull request merged**